compair/static/modules/course/course-module.js

```diff
@@ -346,14 +352,60 @@ module.controller(
                         return !assignment.available;
                     // STUDENTS: return all assignments that need to be answered or compared
                     case "My unfinished assignments":
-                        return (assignment.answer_period && !assignment.status.answers.answered) ||
-                            (assignment.compare_period && assignment.steps_left > 0);
+                        return (permissions.canAnswer && permissions.needsAnswer) ||
+                            (permissions.canCompare && permissions.needsCompareOrSelfEval);
+                    // STUDENTS: return all assignments that have a saved draft answer, comparison, or self-eval
+                    case "Assignments with drafts":
+                        return (permissions.canAnswer && permissions.needsAnswer && permissions.hasDraftAnswer) ||
+                            (permissions.canCompare && permissions.needsCompare && permissions.hasDraftComparison) ||
+                            (permissions.canCompare && permissions.needsSelfEval && permissions.hasDraftSelfEval);
+                    // STUDENTS: return all assignments that have received feedback
+                    case "Assignments with feedback":
+                        return permissions.hasFeedback;
                     default:
                         return false;
                 }
             }
         }
 
+        $scope.assignmentSortOrder = function(sortOrder) {
+            switch (sortOrder) {
+                case "Answer due date":
+                    return function(assignment) {
+                        // return rather undefined if empty, otherwise answer_end if it occurs in the future
+                        if (moment(assignment.answer_end).diff(moment()) > 0) {
+                            return moment(assignment.answer_end).valueOf();
+                        }
+                        return undefined;
+                    };
+                    break;
+                case "Comparisons due date":
+                    return function(assignment) {
+                        // return rather undefined if empty, otherwise compare_end if it occurs in the future
+                        if (assignment.compare_end && moment(assignment.compare_end).diff(moment()) > 0) {
+                            return moment(assignment.compare_end).valueOf();
+                        }
+                        else if (!assignment.compare_end && moment(assignment.answer_end).diff(moment()) > 0) {
+                            return moment(assignment.answer_end).valueOf();
+                        }
+                        return undefined;
+                    };
+                    break;
+                case "Assignment start date":
+                default:
+                    return function(assignment) {
```

Comment:
@andrew-gardener this change makes the tests fail, I think because answer_start and created are treated as text, so it does not sort probably. I'm not sure how we can make those tests pass.

Reply:
I haven't tried it out before but maybe
```
return function(assignment) {
    return [new Date(assignment.answer_start), new Date(assignment.created)];
}
```
or something similar will work

Reply:
Done (had to modify your version a bit)

Reply:
ok please use `return - moment(assignment.answer_start).valueOf();` instead (with the change to orderby in the template, this can be simplified)